modules/gateway/gateway.go

```diff
@@ -159,18 +144,26 @@ func New(addr string, persistDir string) (g *Gateway, err error) {
 	g.log.Println("INFO: gateway created, started logging")
 
 	// Forward the RPC port, if possible.
-	go g.forwardPort(port)
+	g.threads.Add()
+	go func() {
+		defer g.threads.Done()
+		g.forwardPort(port)
```

Comment:
I thought unexported methods that don't begin with `threaded` or `managed` do **not** hold a lock.